### ISSUE TITLE
some children of RefrigerationSystem can be added several times and produce a fatal when System is removed

### ISSUE THREAD

**Issue by jmarrec** [NREL/OpenStudio#3921]
Issue overview
--------------

As part of #3919 (and initial fix in #3920), I'm finding other problems as I go through.

eg: RefrigerationCompressor for eg can be added several times on the the same RefrigerationSystem (and on multiple refrigeration systems). On a single system it can even be assigned on two different `ModelObjectList` (compressorList and highStageCompressorList)

When the `RefrigerationSystem::remove() `is called, it will delete the `ModelObjectList` which:
1. will try to delete the compressor several times produces a crash
2. will (if it works) as a result make any other RefrigerationSystem that has that compressor just lose it which I find confusing

For 1. 
```
m = Model.new; r1 = RefrigerationSystem.new(m); c = RefrigerationCompressor.new(m); r1.addCompressor(c); r1.addCompressor(c);
[22] EnergyPlus(main)> r1.remove
[openstudio.model.RefrigerationCompressor] <2> Object of type 'OS:Refrigeration:Compressor' and named 'Refrigeration Compressor 1' does not have an Refrigeration Compressor Power Curve attached.
RuntimeError: /home/julien/Software/Others/OpenStudio/src/model/RefrigerationCompressor.cpp@134 : Object of type 'OS:Refrigeration:Compressor' and named 'Refrigeration Compressor 1' does not have an Refrigeration Compressor Power Curve attached.
from (pry):22:in `remove'
```

For 2.

```
[23] EnergyPlus(main)> m = Model.new; r1 = RefrigerationSystem.new(m); r2 = RefrigerationSystem.new(m); c = RefrigerationCompressor.new(m); r1.addCompressor(c); r2.addCompressor(c);
[24] EnergyPlus(main)> r1.remove
=> [#<OpenStudio::IdfObject:0x000055f0a602d7e8 @__swigtype__="_p_openstudio__IdfObject">,
 #<OpenStudio::IdfObject:0x000055f0a602d770 @__swigtype__="_p_openstudio__IdfObject">,
 #<OpenStudio::IdfObject:0x000055f0a602d6f8 @__swigtype__="_p_openstudio__IdfObject">,
 #<OpenStudio::IdfObject:0x000055f0a602d6a8 @__swigtype__="_p_openstudio__IdfObject">,
 #<OpenStudio::IdfObject:0x000055f0a602d630 @__swigtype__="_p_openstudio__IdfObject">,
 #<OpenStudio::IdfObject:0x000055f0a602d5b8 @__swigtype__="_p_openstudio__IdfObject">]
[25] EnergyPlus(main)> puts r2.getTarget(5).get.to_ModelObjectList.get
OS:ModelObjectList,
  {4f09a4ca-5ae8-4b1e-ba45-5fbfb194236c}, !- Handle
  Refrigeration System 2 Compressor List, !- Name
  ;                                       !- Model Object 1

```

### Possible Solution

Not sure yet. Possible options:

1. Make the assignment unique, like it is done for Case/Walkins by removing the child from any RefrigerationSystem it already belongs to https://github.com/NREL/OpenStudio/blob/d05bc3a9ba090cf986cde56b7616b7dc445ac2f1/src/model/RefrigerationSystem.cpp#L443-L450
2. Make the child (eg RefrigerationCompressor) a ResourceObject... (... with all that implies....)
3. Manually do a couple of fixes to avoid removing stuff twice (ductape)

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version): all
 - Version of OpenStudio (if using an intermediate build, include SHA): 3.0.0-beta @ d05bc3a

**Comment by jmarrec** [comment]
@kbenne I would like your opinion on this.

**Comment by kbenne** [comment]
I like option 1.

**Comment by tijcolem** [comment]
@jmarrec Looks like we should go with option 1 on this. This is tagged for 3.1.0 release so just wanted to follow up on this.

**Comment by jmarrec** [comment]
@kbenne Actually, I implemented the systems() method for these three in #3920 and it's only been released in 3.0.1... I would say we do not leave the `OS_DEPRECATED systems()`. I doubt anyone has used it, and we'll have to carry it for several versions...

**Comment by kbenne** [comment]
ok. I guess then `systems()` is gone.
